engine/overworld/wild_mons.asm

```diff
@@ -1,21 +1,19 @@
 LoadWildData::
 	ld hl, WildDataPointers
 	ld a, [wCurMap]
-
-	; get wild data for current map
 	ld c, a
 	ld b, 0
 	add hl, bc
 	add hl, bc
 	ld a, [hli]
 	ld h, [hl]
-	ld l, a       ; hl now points to wild data for current map
+	ld l, a
 	ld a, [hli]
-	ld [wGrassRate], a
+	ld [wGrassRate], a ; same as [wLinkEnemyTrainerName]
```

Comment:
Because of the Cinnabar/Old Man glitch, I understand making some cross references between `wGrassRate` and `wLinkEnemyTrainerName` in some spots, but this straightforward routine doesn't feel like the right place. (_This_ code isn't where any of the problems stem from.)